blockchain/thresholdstate.go

```diff
@@ -309,14 +309,28 @@ func (b *BlockChain) thresholdState(prevNode *blockNode, checker thresholdCondit
 //
 // This function is safe for concurrent access.
 func (b *BlockChain) ThresholdState(deploymentID uint32) (ThresholdState, error) {
+	b.chainLock.Lock()
+	state, err := b.deploymentState(b.bestNode, deploymentID)
+	b.chainLock.Unlock()
+
+	return state, err
+}
+
+// depolymentState returns the current rule change threshold for a given
+// deploymentID. The threshold is evaluated from the point of view of the
+// blockNode passed in as the first argument to this method.
+//
+// This function MUST be called with the chain state lock held (for writes).
+func (b *BlockChain) deploymentState(blockNode *blockNode,
```

Comment:
The argument here really should be named `prevNode` because `thresholdState` accepts a prevNode.  That is to say it returns the threshold state for the block _after_ the provided block node (hence the name `prevNode`).

I would also suggest the following diff to make the comments call this out more clearly.

```diff
 // deploymentState returns the current rule change threshold for a given
 // deploymentID. The threshold is evaluated from the point of view of the
-// blockNode passed in as the first argument to this method.
+// block node passed in as the first argument to this method.
+//
+// It is important to note that, as the variable name indicates, this function
+// expects the block node prior to the block for which the deployment state is
+// desired.  In other words, the returned deployment state is for the block
+// AFTER the passed node.
 //
 // This function MUST be called with the chain state lock held (for writes).
-func (b *BlockChain) deploymentState(blockNode *blockNode,
-       deploymentID uint32) (ThresholdState, error) {
-
+func (b *BlockChain) deploymentState(prevNode *blockNode, deploymentID uint32) (ThresholdState, error) {
```